dbus/xdg.c

```diff
@@ -93,23 +87,32 @@ static int handle_notify(sd_bus_message *msg, void *data,
 		return ret;
 	}
 
-	free(notif->app_name);
-	free(notif->app_icon);
-	free(notif->summary);
-	free(notif->body);
+	struct mako_notification *notif = NULL;
+	if (replaces_id > 0) {
+		notif = get_notification(state, replaces_id);
+	}
+
+	if (notif) {
+		reset_notification(notif);
+	} else {
+		// Either we had no replaces_id, or the id given was invalid. Either
+		// way, make a new notification.
```

Comment:
Maybe we can error out instead if the ID is invalid?

Reply:
The spec doesn't have anything to say about this, so I thought it might be better to err on the side of not punishing a client that misses replacing it by just a moment. For example, the `notify-send` replacement that I use tracks its replacement id in a file, which is not super robust but works for a client that doesn't run continuously.

Reply:
That's right, I agree.